core-bundle/contao/library/Contao/System.php

```diff
@@ -592,6 +594,7 @@ public static function setCookie($strName, $varValue, $intExpires, $strPath=null
 		{
 			foreach ($GLOBALS['TL_HOOKS']['setCookie'] as $callback)
 			{
+				trigger_deprecation('contao/core-bundle', '5.3', 'Using the "setCookie" hook has been deprecated and will no longer work in Contao 6. Use kernel.response events instead.');
```

Comment:
Should we move this to line 594 before the `foreach` loop? There is no need to trigger the generic deprecation message multiple times, is there?